http/encoding.go

```diff
@@ -25,6 +27,13 @@ const (
 	ContentTypeKey
 )
 
+const (
+	// ErrUnsupportedMediaTypeName is the name of the Goa service error returned
+	// by the built-in decoder when the content type of the request body is not
+	// supported.
+	ErrUnsupportedMediaTypeName = "ErrUnsupportedMediaType"
+)
+
```

Comment:
Should it be defined in `pkg/error.go`?

```go
const (
	UnsupportedMediaType = "unsupported_media_type"
)

func UnsupportedMediaTypeError(ct string) error {
	return PermanentError(UnsupportedMediaType, "unsupported media type %s", ct)
}
```

It seems better to define the error name in snake case like the existing definitions.

* `missing_payload`
* `decode_payload`
* `invalid_field_type`
* `missing_field`
* `invalid_enum_value`
* `invalid_format`
* `invalid_pattern`
* `invalid_range`
* `invalid_length`

Reply:
Thank you for the comment! Yes I agree that the name should be snake case for consistency. `pkg/error.go` contains service level errors while this is a HTTP transport level error so it might be better to keep this error definition in the `http` package.

Reply:
Makes sense, I moved the error definition to `pkg/error.go`.